### PR TITLE
MINOR: add github action trigger

### PR DESCRIPTION
**Pull request by waynexia**
Signed-off-by: Ruihang Xia <waynestxia@gmail.com>

# Which issue does this PR close?

Ref #3321.

 # Rationale for this change

GitHub Action will only run on PR updating, we cannot trigger one on demand manually. This is inconvenient if we want to run one on master or without submitting a PR.

I've tried this patch in my fork repo, now I can see this button:
image

# What changes are included in this PR?

Add an [action trigger](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_dispatch] that allows us to trigger an action manually.

# Are there any user-facing changes?

No

### CODE REVIEW

**Comment by andygrove** [comment]
test failure caused by https://github.com/apache/arrow-datafusion/issues/3327

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -18,9 +18,11 @@
 name: Rust
 
 on:
-  # always trigger
+  # always trigger on PR
   push:
   pull_request:
+  # manual trigger
+  workflow_dispatch:
```
Comment (alamb) [inline .github/workflows/rust.yml:25]: TIL https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_dispatch

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -18,9 +18,11 @@
 name: Rust
 
 on:
-  # always trigger
+  # always trigger on PR
   push:
   pull_request:
+  # manual trigger
```
Comment (alamb) [inline .github/workflows/rust.yml:24]:
```suggestion
  # manual trigger
  # https://docs.github.com/en/actions/managing-workflow-runs/manually-running-a-workflow
```

**Review by alamb — APPROVED** [review]

**Comment by alamb** [comment]
Thanks @waynexia  -- this looks great.

**Comment by alamb** [comment]
I took the liberty of merging this branch to master -- hope to merge it when green

**Comment by waynexia** [comment]
Thanks ❤️

**Pull request merged**